Iris 1289 - java, js native fps (#3025)
saucelabs/sauce-docs: docs/visual-testing/integrations/java.md

By default, only the current viewport is captured when `.sauceVisualCheck` is used. You can opt in to capturing the entire page by using the `enableFullPageScreenshots` option. It will capture everything by scrolling and stitching multiple screenshots together.

Configuration should be specified using the `FullPageScreenshotConfig.Builder` object.

<FullPageLimit />

#### Web

> Methods available:

Reviewer: ```suggestion
Available methods:
```

CheckOptions options = new CheckOptions();
FullPageScreenshotConfig config = new FullPageScreenshotConfig.Builder()
    .withDelayAfterScrollMs(500)
    .withDisableCSSAnimation(false)
    .withHideAfterFirstScroll("#header")
    .withHideScrollBars(false)
    .withScrollLimit(5)
    .build();
options.enableFullPageScreenshots(config);
visual.sauceVisualCheck("Long content page", options);
```

#### Mobile Native (beta)

Methods available:
- `withDelayAfterScrollMs(int delayAfterScrollMs)`: Delay in ms after scrolling and before taking screenshots. The default value is 0. We recommend using this option for lazy loading content.
- `withNativeClipSelector(SelectorIn nativeClipSelector)`: Selector used to identify the first element to which clipping will be applied.
- `withScrollElement(WebElement scrollElement)`: Scrollable element used for scrolling. The default is root element.
- `withScrollLimit(int scrollLimit)`: Limit the number of screenshots taken for scrolling and stitching. The default value is 10. The value needs to be between 1 and 10.

:::note

> It is recommended to define the `withScrollElement` as the appropriate scrollable container.

Reviewer: ```suggestion
It is recommended to use the `withScrollElement` method to set the appropriate scrollable container.
```

:::

> 

Reviewer: When looking to the rendered doc (https://docs.dev.saucelabs.net/pr-preview/pr-3025/visual-testing/integrations/java/#mobile-native-beta), this is not super clear what are those 3 blocks of code.

It took me several seconds to understand that this was actually different examples, with different configurations. We should definitively add text explaining what a customer is seeing and/or a title on top of it.

Otherwise that's probably not clear enough.

> <Tabs>

Reviewer: ```suggestion
Examples:

<Tabs>
```

        CheckOptions options = new CheckOptions();
        FullPageScreenshotConfig config = new FullPageScreenshotConfig.Builder()
            .withScrollElement(scrollElement)
            .withScrollLimit(5)
            .build();
        options.enableFullPageScreenshots(config);
        visual.sauceVisualCheck("Long content page", options);
        ```
    </TabItem>
    <TabItem value="android" label="Android">
        ```java
        import com.saucelabs.visual.CheckOptions;
        import com.saucelabs.visual.model.FullPageScreenshotConfig;
        
        RemoteWebDriver driver;
        ...
        
        WebElement scrollElement = driver.findElement(AppiumBy.xpath("//androidx.recyclerview.widget.RecyclerView")); 
        CheckOptions options = new CheckOptions();
        FullPageScreenshotConfig config = new FullPageScreenshotConfig.Builder()
            .withScrollElement(scrollElement)
            .withScrollLimit(5)
            .build();
        options.enableFullPageScreenshots(config);
        visual.sauceVisualCheck("Long content page", options);
        ```
    </TabItem>
</Tabs>

> Use only XPath selectors for ignore regions and clipping to an element.

Reviewer: ```suggestion
Only XPath selectors can be used for ignore regions and clipping to an element.
```

> :::note
> On iOS, selectors must be contained within the `scrollElement`.
> :::

Reviewer: I'd put this into a warning rather than a note. IMO it needs to be highlighted.

```suggestion
:::warning
On iOS, selectors **must** be contained within the `scrollElement`.
:::
```
